src/inspector/models/score/scoreaccessibilitysettingsmodel.cpp

```diff
@@ -0,0 +1,179 @@
+        if (!m_ignoreStyleChange) {
+            isAccessibilityStyleEdited = true;
+            globalContext()->currentNotation()->elements()->msScore()->score()->style().setStylePresetEdited(true);
+            emit possibleScoreStylePresetsChanged();
+            emit scoreStylePresetIndexChanged();
+        }
+    });
+}
+
+void ScoreAccessibilitySettingsModel::createProperties()
+{
+    // Placeholder
+}
+
+void ScoreAccessibilitySettingsModel::requestElements()
+{
+    // Placeholder
+}
+
+void ScoreAccessibilitySettingsModel::loadProperties()
+{
+    // Placeholder
+}
+
+void ScoreAccessibilitySettingsModel::resetProperties()
+{
+    // Placeholder
+}
+
+muse::io::path_t ScoreAccessibilitySettingsModel::getStyleFilePath(ScoreStylePreset preset) const
+{
+    muse::io::path_t basePath = globalConfiguration()->appDataPath() + "styles/MSN/";
+    switch (preset) {
+    case ScoreStylePreset::DEFAULT:
+        return engravingConfiguration()->defaultStyleFilePath();
+    case ScoreStylePreset::MSN_16MM:
+        return basePath + "16mm_MSN.mss";
+    case ScoreStylePreset::MSN_18MM:
+        return basePath + "18mm_MSN.mss";
+    case ScoreStylePreset::MSN_20MM:
+        return basePath + "20mm_MSN.mss";
+    case ScoreStylePreset::MSN_22MM:
+        return basePath + "22mm_MSN.mss";
+    case ScoreStylePreset::MSN_25MM:
+        return basePath + "25mm_MSN.mss";
+    default:
+        return muse::io::path_t();
+    }
+}
+
+QVariantList ScoreAccessibilitySettingsModel::possibleScoreStylePresets() const
+{
+    QVariantList presets;
+    MStyle style = globalContext()->currentNotation()->elements()->msScore()->score()->style();
+    int currentPresetValue = static_cast<int>(style.stylePreset());
+
+    QString text = "text";
+    QString value = "value";
+    QString preset = "preset";
+    QString edited = "edited";
+
+    for (int i = 0; i < static_cast<int>(ScoreStylePreset::MAX_PRESET); ++i) {
+        ScoreStylePreset presetEnum = static_cast<ScoreStylePreset>(i);
+        QString presetName = TConv::translatedUserName(presetEnum);
+
+        presets.append(QVariantMap {
+            { text, presetName },
+            {
+                value,
+                QVariantMap {
+                    { preset, i },
+                    { edited, false }
+                }
+            }
+        });
+
+        if (i == currentPresetValue && style.stylePresetEdited()) {
+            QString editedPresetName = muse::qtrc("inspector", "%1 (edited)").arg(presetName);
+            presets.append(QVariantMap {
+                { text, editedPresetName },
+                {
+                    value,
+                    QVariantMap {
+                        { preset, i },
+                        { edited, true }
+                    }
+                }
+            });
+        }
+    }
+
+    return presets;
+}
+
+void ScoreAccessibilitySettingsModel::setScoreStylePresetIndex(int index)
+{
+    loadStyle(index);
+}
```

Comment:
Here you're supposed to use the supplied `int index` to determine whether the preset enum and/or edited value has changed. If so, you should update those values in the style (as well as your cached versions if caching) and then emit the appropriate signals.

Reply:
Yes, that makes sense. I thought about it like this:

As we know, the member variables `m_scoreStylePreset` and `m_scoreStylePresetEdited` are caching the latest `ScoreStylePreset` and whether it is an edited version or not. We can use this information to determine if there has been a change. When `m_scoreStylePresetEdited` is `false`, it means there are 6 values in the dropdown; otherwise, there are 7 values, with the edited option always appearing just below the corresponding standard option. So it should look like the following:

```cpp
void ScoreAccessibilitySettingsModel::setScoreStylePresetIndex(int index)
{
    bool isEdited = false;
    ScoreStylePreset selectedPreset;

    if (m_scoreStylePresetEdited) {
        if (index == static_cast<int>(m_scoreStylePreset) + 1) {
            isEdited = true;
        }
        selectedPreset = static_cast<ScoreStylePreset>(index - (isEdited ? 1 : 0));
    } else {
        selectedPreset = static_cast<ScoreStylePreset>(index);
    }

    bool presetChanged = (selectedPreset != m_scoreStylePreset);
    bool editedChanged = (isEdited != m_scoreStylePresetEdited);

    if (presetChanged || editedChanged) {
        m_scoreStylePreset = newPreset;
        m_scoreStylePresetEdited = isEdited;

        if (!isEdited) {
            loadStyle(static_cast<int>(m_scoreStylePreset));
        }

        emit possibleScoreStylePresetsChanged();
        emit scoreStylePresetIndexChanged();
    }
}
```
I think the signal `possibleScoreStylePresetsChanged()` should ideally be emitted when there is a selection  from edited version to standard and not from standard to standard but yeah for simplicity, I have kept it like this for now. I just updated the PR with the updates :)